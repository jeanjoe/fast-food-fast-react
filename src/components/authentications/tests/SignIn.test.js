import React from "react";
import expect from "expect";
import Adapter from "enzyme-adapter-react-16";
import { shallow, configure } from "enzyme";
import { SignIn } from "../SignIn";
configure({ adapter: new Adapter() });

describe("Test article component", () => {
  const props = {
    error: [],
    user: [],
    loginUser: jest.fn()
  };
  const wrapper = shallow(<SignIn {...props} />);

  it("Matches the snapshot", () => {
    expect(wrapper).toMatchSnapshot();
  });
  it("Handle handleOnChange", () => {
    const event = {
      target: {
        value: "text"
      }
    };
    wrapper.instance().handleOnChange(event);
    expect(wrapper.state("loginError")).toEqual(null);
  });
  it("Handle handleOnChange", () => {
    wrapper.instance().handleSubmit();
    expect(props.loginUser).toHaveBeenCalled();
  });
  it("Handle componentWillReceiveProps", () => {
    const nextProps = {
      user: {
        message: "Login successfully"
      },
      error: {}
    };
    wrapper.instance().componentWillReceiveProps(nextProps);
    expect(wrapper.state("loginSuccess")).toEqual(
      "Login successful, Redirecting you..."
    );
  });
  it("Handle componentWillReceiveProps on Error", () => {
    const nextProps = {
      user: { },
      error: {
          response: {
              data: 'error message'
          }
      }
    };
    wrapper.instance().componentWillReceiveProps(nextProps);
    expect(wrapper.state("loginError")).toEqual(
        'error message'
    );
  });
});
