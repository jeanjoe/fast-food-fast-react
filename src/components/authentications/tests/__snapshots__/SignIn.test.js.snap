// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test article component Matches the snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <SignIn
    error={Array []}
    loginUser={[MockFunction]}
    user={Array []}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": <Grid
        centered={true}
        stackable={true}
      >
        <GridColumn
          width={6}
        >
          <Segment
            style={
              Object {
                "margin": "5em 0em",
              }
            }
          >
            <SignInForm
              handleOnChange={[Function]}
              handleSubmit={[Function]}
              loading={false}
              loginError={null}
            />
          </Segment>
        </GridColumn>
      </Grid>,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "centered": true,
        "children": <GridColumn
          width={6}
        >
          <Segment
            style={
              Object {
                "margin": "5em 0em",
              }
            }
          >
            <SignInForm
              handleOnChange={[Function]}
              handleSubmit={[Function]}
              loading={false}
              loginError={null}
            />
          </Segment>
        </GridColumn>,
        "stackable": true,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "children": <Segment
            style={
              Object {
                "margin": "5em 0em",
              }
            }
          >
            <SignInForm
              handleOnChange={[Function]}
              handleSubmit={[Function]}
              loading={false}
              loginError={null}
            />
          </Segment>,
          "width": 6,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": <SignInForm
              handleOnChange={[Function]}
              handleSubmit={[Function]}
              loading={false}
              loginError={null}
            />,
            "style": Object {
              "margin": "5em 0em",
            },
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "handleOnChange": [Function],
              "handleSubmit": [Function],
              "loading": false,
              "loginError": null,
              "loginSuccess": undefined,
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": [Function],
        },
        "type": [Function],
      },
      "type": [Function],
    },
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": <Grid
          centered={true}
          stackable={true}
        >
          <GridColumn
            width={6}
          >
            <Segment
              style={
                Object {
                  "margin": "5em 0em",
                }
              }
            >
              <SignInForm
                handleOnChange={[Function]}
                handleSubmit={[Function]}
                loading={false}
                loginError={null}
              />
            </Segment>
          </GridColumn>
        </Grid>,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "centered": true,
          "children": <GridColumn
            width={6}
          >
            <Segment
              style={
                Object {
                  "margin": "5em 0em",
                }
              }
            >
              <SignInForm
                handleOnChange={[Function]}
                handleSubmit={[Function]}
                loading={false}
                loginError={null}
              />
            </Segment>
          </GridColumn>,
          "stackable": true,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": <Segment
              style={
                Object {
                  "margin": "5em 0em",
                }
              }
            >
              <SignInForm
                handleOnChange={[Function]}
                handleSubmit={[Function]}
                loading={false}
                loginError={null}
              />
            </Segment>,
            "width": 6,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "children": <SignInForm
                handleOnChange={[Function]}
                handleSubmit={[Function]}
                loading={false}
                loginError={null}
              />,
              "style": Object {
                "margin": "5em 0em",
              },
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "handleOnChange": [Function],
                "handleSubmit": [Function],
                "loading": false,
                "loginError": null,
                "loginSuccess": undefined,
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            "type": [Function],
          },
          "type": [Function],
        },
        "type": [Function],
      },
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
  Symbol(enzyme.__childContext__): null,
}
`;
